tests/test_poincare_half_space.py

```diff
@@ -25,6 +25,14 @@ def test_belongs(self):
         expected = gs.array([True, False])
         self.assertAllClose(result, expected)
 
+    def test_inner_product_vectorization(self):
```

Comment:
Did you try other vectorization scenarios? E.g. one pair of tangent_vec + one base_point, one tangant_vec_a, multiple tangent_vec_b, one base_point, or multiple pairs of tangent_vecs at one base_point?

Reply:
Yes they also work. Should I include them in ```test_poincare_half_space```?